sycl/include/sycl/detail/common.hpp

```diff
@@ -173,14 +173,13 @@ inline std::string codeToString(pi_int32 code) {
                           "Native API returns: "
 
 #ifndef __SYCL_SUPPRESS_PI_ERROR_REPORT
-#include <sycl/detail/iostream_proxy.hpp>
 // TODO: rename all names with direct use of OCL/OPENCL to be backend agnostic.
 #define __SYCL_REPORT_PI_ERR_TO_STREAM(expr)                                   \
   {                                                                            \
     auto code = expr;                                                          \
     if (code != PI_SUCCESS) {                                                  \
-      std::cerr << __SYCL_PI_ERROR_REPORT << sycl::detail::codeToString(code)  \
-                << std::endl;                                                  \
+      fprintf(stderr, __SYCL_PI_ERROR_REPORT "%s\n",                           \
+              sycl::detail::codeToString(code).c_str());                       \
```

Comment:
To make the code equivalent to previous version, we need to call fflush.

The same applies to sycl/include/sycl/handler.hpp.